### PR TITLE
Reduce getAccountInfo() calls

### PR DESCRIPTION
**Pull request by nathanleclaire**
Closes https://github.com/workbenchapp/solana-workbench/issues/118

### CODE REVIEW

**Comment by nathanleclaire** [comment]
so happy with how far the performance and stability has come btw @SvenDowideit !

Screen Shot 2022-05-06 at 2 42 50 PM

File: src/renderer/components/ProgramChangeView.tsx [hunk 1]
```diff
@@ -9,6 +9,7 @@ import { toast } from 'react-toastify';
 
 import OutsideClickHandler from 'react-outside-click-handler';
 
+import { Account } from '@solana/web3.js';
```
Comment (SvenDowideit) [inline src/renderer/components/ProgramChangeView.tsx:12]: boooooo to the linties :)

**Comment by SvenDowideit** [comment]
ooooo, i was going to object, because i wanted to see updates to starred accounts that might happen outside the subscription, but actually, yeah - this works for me!

**Review by SvenDowideit — APPROVED** [review]
other than the lintie, lgtm!

**Comment by SvenDowideit** [comment]
mmm, ok, after more testing - no, this doesn't work - when you have starred accounts on a chatty network, they don't stay at the top - they get lost

**Review by SvenDowideit — CHANGES_REQUESTED** [review]
this breaks the starred account functionality

while I agree that we don't need to poll starred accounts for their info, we can use the stale one from the cache, this makes it impossible to watch the accounts I selected.

**Comment by SvenDowideit** [comment]
So I think this is not a trivial fix to make - it basically needs a way to use the account info cache with stale info for starred accounts

**Comment by nathanleclaire** [comment]
> this breaks the starred account functionality

Ah womp, I'll look into it ... perhaps 0.3.1

**Comment by SvenDowideit** [comment]
#86 introduces ux to allow the user to choose what column to sort by, and moved the sorting to the UI, which also dramatically reduces the number of uncached getAccount requests - so __hopefully__ that PR can replace this one :)

**Comment by SvenDowideit** [comment]
mmmm, or not - as soon as i request some airdrops, I get further issues - so yup, carry on :)

**Comment by nathanleclaire** [comment]
Hm, I'm uncertain on the best solution to move forward with here. Ultimately, pinned accounts and program changes don't really have a good way of knowing about each other (should they?), and that makes this tricky.

One idea I had was the render the pinnedAccounts first, before the changes `map`, but it is simply a list of strings, so we have no way to display the other info easily in the UI. (Maybe I'll make the `getAccount` poll optional for now?) Likewise changes subscription logic also does not know about pinned accounts, so it can't bubble them to the top in its sort.

**Comment by SvenDowideit** [comment]
the getAccount poll _is_ optional already - that's why its just a cache call - the cache is filled from program changes. (and we should only go out to the rpc server if the subscription hasn't updated the info recently enough (the changes i made in the most recent update to #86 actually just encourage the cache to always give us stale cached info from the program changes source.

**Comment by nathanleclaire** [comment]
so, here's my proposal for now ... 

split pinned accounts and changes into their own sections... example below

https://user-images.githubusercontent.com/1476820/167509520-3e2019ef-aba6-4d3d-8803-f49f4fd252d8.mov

this is a lot closer to good to go now, pinned accounts are back ... rough edges when switching nets... i spotted a bug, if you switch localnet -> main -> localnet, it looks like changes no longer update :(

but this is a lot closer

**Pull request closed without merging**